### PR TITLE
CLDC-3019 Allow scheme editing for support users

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Allow support users edit more schemes and locations details after the resources have been created.

### CODE REVIEW

File: app/views/schemes/check_answers.html.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
       <h2 class="govuk-visually-hidden">Scheme</h2>
       <dl class="govuk-summary-list">
         <% @scheme.check_details_attributes.each do |attr| %>
-          <%= render partial: "scheme_summary_list_row", locals: { scheme: @scheme, attribute: attr, change_link: @scheme.confirmed? ? scheme_edit_name_path(@scheme) : scheme_details_path(@scheme, check_answers: true) } %>
+          <%= render partial: "scheme_summary_list_row", locals: { scheme: @scheme, attribute: attr, change_link: current_user.support? ? scheme_details_path(@scheme, check_answers: true) : scheme_edit_name_path(@scheme) } unless attr[:name] == "Status" %>
```
Comment (natdeanlewissoftwire) [inline app/views/schemes/check_answers.html.erb]: won't this redirect to the edit name path for an unconfirmed (incomplete) scheme viewed by a data coordinator? I think we should still reirect to the details path in that case

File: app/views/schemes/show.html.erb [hunk 2]
```diff
@@ -14,24 +14,48 @@
...
-          <% row.value do %>
-            <%= details_html(attr) %>
-            <% if attr[:name] == "Status" && @scheme.confirmed? && @scheme.locations.confirmed.none? && LocationPolicy.new(current_user, @scheme.locations.new).create? %>
-              <span class="app-!-colour-muted">Complete this scheme by adding a location using the <%= govuk_link_to("‘locations’ tab", scheme_locations_path(@scheme)) %>.</span>
-            <% end %>
-          <% end %>
-          <% if SchemePolicy.new(current_user, @scheme).update? %>
-            <% row.action(text: "Change", href: scheme_edit_name_path(scheme_id: @scheme.id)) if attr[:edit] %>
-          <% end %>
+    <dl class="govuk-summary-list">
```
Comment (natdeanlewissoftwire) [inline app/views/schemes/show.html.erb:22]: nice :D

File: app/views/schemes/show.html.erb [hunk 3]
```diff
@@ -14,24 +14,48 @@
...
-      <% display_scheme_attributes(@scheme).each do |attr| %>
-        <%= summary_list.row do |row| %>
-          <% row.key { attr[:name] } %>
-          <% row.value do %>
-            <%= details_html(attr) %>
-            <% if attr[:name] == "Status" && @scheme.confirmed? && @scheme.locations.confirmed.none? && LocationPolicy.new(current_user, @scheme.locations.new).create? %>
-              <span class="app-!-colour-muted">Complete this scheme by adding a location using the <%= govuk_link_to("‘locations’ tab", scheme_locations_path(@scheme)) %>.</span>
-            <% end %>
-          <% end %>
-          <% if SchemePolicy.new(current_user, @scheme).update? %>
```
Comment (natdeanlewissoftwire) [inline app/views/schemes/show.html.erb:34]: is the scheme update policy still needed? or should it be changed at all in the light of this PR?

File: app/models/scheme.rb [hunk 4]
```diff
@@ -187,37 +187,38 @@ def id_to_display
   def check_details_attributes
```
Comment (natdeanlewissoftwire) [inline app/models/scheme.rb:187]: do we always use any of these `check_..._attributes` methods together? can we not combine any of them any further?

**Review by natdeanlewissoftwire — COMMENTED** [review]
looks good! one general thing - I can't see the inset text in the designs anywhere here, is that being included in a different PR?
image

File: app/views/schemes/check_answers.html.erb (hunk 1 above)
Comment (kosiakkatrina) [reply app/views/schemes/check_answers.html.erb]: Good spot!

File: app/views/schemes/show.html.erb (hunk 3 above)
Comment (kosiakkatrina) [reply app/views/schemes/show.html.erb:34]: I think at this point the policy is more around whether the user is someone who should be able to update the scheme (whether they're in the right org/role) as opposed to the state of the scheme, so I think it's still worth keeping around

File: app/models/scheme.rb (hunk 4 above)
Comment (kosiakkatrina) [reply app/models/scheme.rb:187]: Yup! Refactored it

**Comment by kosiakkatrina** [comment]
> looks good! one general thing - I can't see the inset text in the designs anywhere here, is that being included in a different PR? 

@natdeanlewissoftwire ah, nope I just missed it, thanks! It's there now!

**Review by natdeanlewissoftwire — APPROVED** [review]
looks good!

**Pull request merged**